pkg/service/service.go

```diff
@@ -145,9 +144,16 @@ func (s *Service) Run() error {
 			if !s.killed.Load() {
 				activeCalls := s.sipServiceActiveCalls()
 				if activeCalls > 0 {
-					fmt.Printf("instance waiting for %d calls to finish", activeCalls)
-					time.Sleep(shutdownTimer)
-					continue
+					s.log.Infow("waiting for calls to finish", "calls", activeCalls)
+					shutdownTicker := time.NewTicker(5 * time.Second)
+					for {
+						<-shutdownTicker.C
+						activeCalls = s.sipServiceActiveCalls()
+						if activeCalls == 0 {
+							break
+						}
+						s.log.Infow("waiting for calls to finish", "calls", activeCalls)
+					}
 				}
 			}
```

Comment:
Good catch! I think the outer loop was supposed to do this, but it was lost in the initial refactor, probably :thinking: 

Now in the new version the inner loop will need to check for `s.killed` as well (if we decide to kill it after graceful shutdown was started). Also a good idea to  `defer shutdownTicker.Stop()`.

Maybe something like this?

```go
	if !s.killed.Load() {
	      shutdownTicker := time.NewTicker(5 * time.Second)
	      defer shutdownTicker.Stop()
	      for !s.killed.Load() {
		      activeCalls = s.sipServiceActiveCalls()
		      if activeCalls == 0 {
			      break
		      }
		      s.log.Infow("waiting for calls to finish", "calls", activeCalls)
		      <-shutdownTicker.C
	      }
	}
```

Reply:
yeah good catch/refactor will update!